sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -202,5 +202,9 @@ private[sql] class JacksonGenerator(
    */
   def write(array: ArrayData): Unit = writeArray(writeArrayData(array, arrElementWriter))
 
+  def write(map: MapData, mapType: MapType): Unit = {
+    writeObject(writeMapData(map, mapType, makeWriter(mapType.valueType)))
```

Comment:
I think we could avoid making the writer by `makeWriter(mapType.valueType))` per each map write, by creating this once when this `JacksonGenerator`.